main.go

```diff
@@ -190,17 +211,20 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up health check")
-		os.Exit(1)
-	}
-	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up ready check")
-		os.Exit(1)
-	}
+	leaderElectionContext, cancelLeaderElection := context.WithCancel(context.Background())
+	go func() {
+		setupLog.Info("starting leader election manager")
+		if err := leaderElectionMgr.Start(leaderElectionContext); err != nil {
+			setupLog.Error(err, "Leader Election Manager exited non-zero")
+			os.Exit(1)
+		}
+	}()
+
+	<-leaderElectionMgr.Elected()
 
 	stopCh := ctrl.SetupSignalHandler()
 	go func() {
+		setupLog.Info("starting global manager")
 		if err := mgrGlobal.Start(stopCh); err != nil {
 			setupLog.Error(err, "Manager Global exited non-zero")
 			os.Exit(1)
```

Comment:
shouldn't we cancel the context here as well before exiting?

Reply:
ignore this if defer goes before the shutdown, just after it is elected

Reply:
do you mean `leaderElectionContext`?